### ISSUE TITLE
Support configurable KNOWN_CLAUDE_TOOLS via config

### ISSUE THREAD

**Issue by aRustyDev** [carlrannaberg/cclint#4]
## Summary

`KNOWN_CLAUDE_TOOLS` is a hardcoded Set. There's no config option to register additional tools, so any valid Claude Code tool not in the list produces false-positive warnings with no workaround.

## Proposed

Add an `additionalTools` (or `knownTools`) config option:

```js
// cclint.config.js
export default {
  rules: {
    additionalTools: ['AskUserQuestion', 'EnterPlanMode', 'TaskCreate'],
  },
};
```

This would merge into the built-in set before tool validation runs in both the agents and commands linters.

## Context

Related to #2 — even after adding `AskUserQuestion` to the built-in set, new tools may be added to Claude Code faster than cclint releases. A config escape hatch prevents users from being blocked.
